xml_handlers/OpenCoverTar_handler.php

```diff
@@ -0,0 +1,275 @@
+  the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+  PURPOSE. See the above copyright notices for more information.
+=========================================================================*/
+
+require_once 'xml_handlers/abstract_handler.php';
+require_once 'models/coverage.php';
+require_once 'config/config.php';
+require_once 'models/build.php';
+
+class OpenCoverTarHandler extends AbstractHandler {
+    protected $Build;
+    protected $CoverageSummaries;
+
+    public function __construct($buildid)    {
+        parent::__construct($buildid, $buildid);
+        $this->Build = new Build();
+        $this->Build->Id = $buildid;
+        $this->Build->FillFromId($this->Build->Id);
+
+        $this->CoverageSummaries = array();
+        $coverageSummary = new CoverageSummary();
+        $coverageSummary->BuildId = $this->Build->Id;
+        $this->CoverageSummaries['default'] = $coverageSummary;
+
+        $this->Coverages = array();
+        $this->CoverageFiles = array();
+        $this->CoverageFileLogs = array();
+    }
+
+    public function startElement($parser, $name, $attributes) {
+        parent::startElement($parser, $name, $attributes);
+        /**
+         *  SEQUENCEPOINT denotes a line in the source file that is exectuable and
+         *  may have been executed.
+         *
+         *  VC -> Visit Count
+         *  EL -> Line offset, reduced by one to start the file at line 0
+         */
+        if(($name == "SEQUENCEPOINT") && ($this->coverageFileLog)) {
+            $this->coverageFileLog->AddLine($attributes['EL']-1, $attributes['VC']);
+        }
+    }
+
+    // No usage of endElement
+    public function endElement($parser, $name) {
+    }
+    /**
+     *  Removes the current module, and assumes all that is left is a
+     *  subdirectory and the file name, append '.cs' to get the file path
+     *  in the un-tarred directory
+     **/
+    public function parseFullName($string) {
+        foreach ($this->currentModule as $path) {
+             $filePath = str_ireplace($path,'',$string);
+             $filePath = str_replace('.','/',$filePath);
+             if(file_exists($this->tarDir.'/'.$path.$filePath.'.cs')) {
+                return $path.$filePath.'.cs';
+             }
+        }
+        return false;
+    }
+    // Queries for the coverage objects for both adding source and
+    // adding coverage values
+    public function getCoverageObjects($path) {
+        if (!array_key_exists($path, $this->CoverageFileLogs)) {
+            $coverageFileLog = new CoverageFileLog();
+            $coverageFileLog->BuildId =  $this->Build->Id;
+            $coverageFile = new CoverageFile();
+            $coverageFile->FullPath = trim($path);
+            // Get the ID for this coverage file, or create a new empty one
+            //if it doesn't already exist.
+            $sql = pdo_query(
+            "SELECT id FROM coveragefile
+            WHERE fullpath='$path' AND file IS NULL");
```

Comment:
I lose track of how our pdo implementation works, but is this injecting a filename directly into SQL?